### DISCUSSION TITLE
Including a user setup configuration in an Arduino sketch

### DISCUSSION THREAD

**Discussion by daxliniere** [Bodmer/TFT_eSPI#1521] [Q&A - General] [answered]
Hey again,
Is it possible to override the values in user_setup.h from the main sketch file? I just updated the library and my sketches all stopped working. I realised that user_setup.h might have been replaced during the upgrade and confirmed this is the case.
To avoid future problems, is there a way I can specify override certain settings from my main sketch? Or perhaps make it look for a different user_setup.h file (from main sketch) ?

All the best,
Dax.

**Comment by Strooom** [comment]
I have the same need. I had TFT_eSPI copied as a local library, but I now want to have it as an external library in platformio.
However, the display type, pins definitions etc. are all dedicated to this project, so user_setup.h belongs more to my specific project rather than to the external library.

**Comment by daxliniere** [comment]
May I propose that if a definition is found in the main sketch, it is used instead of what's found in user_setup.h? This way existing projects won't be broken.

**Comment by Bodmer** [comment]
The Arduino IDE environment hides any #defines within sketches from libraries, presumably to prevent chaos from noobs.

In platformIO the normal approach is to use your platformio.ini file as described here:
https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/PlatformIO/Configuring%20options.txt

**Reply by Bodmer** [reply]
Note also that this must not be tampered with!

https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup_Select.h#L16-L19

**Reply by daxliniere** [reply]
> In platformIO the normal approach is to use your platformio.ini file as described here:
> https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/PlatformIO/Configuring%20options.txt

Hey Bodmer, this link is not working, unfortunately. Has it moved somewhere else?

**Comment by Bodmer** [comment]
Yes, it is here now.
https://github.com/Bodmer/TFT_eSPI/tree/master/docs/PlatformIO

**Comment by Bodmer** [comment, marked as answer]
See here for a work-around that allows the setup file to be included in the sketch:

https://github.com/Bodmer/TFT_eSPI/discussions/2007#discussioncomment-3614189
